powertools-large-messages/src/main/java/software/amazon/lambda/powertools/largemessages/internal/LargeMessageProcessor.java

```diff
@@ -0,0 +1,121 @@
+package software.amazon.lambda.powertools.largemessages.internal;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.lambda.powertools.largemessages.LargeMessageConfig;
+import software.amazon.lambda.powertools.largemessages.LargeMessageProcessingException;
+import software.amazon.payloadoffloading.S3BackedPayloadStore;
+import software.amazon.payloadoffloading.S3Dao;
+
+import static java.lang.String.format;
+
+/**
+ * Abstract processor for Large Messages. Handle the download from S3 and replace the actual S3 pointer with the content
+ * of the S3 Object leveraging the payloadoffloading library.
+ *
+ * @param <T> any message type that support Large Messages with S3 pointers
+ *            ({@link com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage} and {@link com.amazonaws.services.lambda.runtime.events.SNSEvent.SNSRecord} at the moment)
+ */
+public abstract class LargeMessageProcessor<T> {
+    protected static final String RESERVED_ATTRIBUTE_NAME = "ExtendedPayloadSize";
+    private static final Logger LOG = LoggerFactory.getLogger(LargeMessageProcessor.class);
+
+    private final S3Client s3Client = LargeMessageConfig.get().getS3Client();
+    private final S3BackedPayloadStore payloadStore = new S3BackedPayloadStore(new S3Dao(s3Client), "DUMMY");
+
+    public Object process(ProceedingJoinPoint pjp, boolean deleteS3Object) throws Throwable {
+        Object[] proceedArgs = pjp.getArgs();
+        T message = (T) proceedArgs[0];
+
+        if (!isLargeMessage(message)) {
+            LOG.warn("Not a large message, proceeding");
+            return pjp.proceed(proceedArgs);
+        }
+
+        String payloadPointer = getMessageContent(message);
+        if (payloadPointer == null) {
+            LOG.warn("No content in the message, proceeding");
+            return pjp.proceed(proceedArgs);
+        }
+        // legacy attribute (sqs only)
+        payloadPointer = payloadPointer.replace("com.amazon.sqs.javamessaging.MessageS3Pointer", "software.amazon.payloadoffloading.PayloadS3Pointer");
+
+        LOG.info("Large message [{}]: retrieving content from S3", getMessageId(message));
+
+        String s3ObjectContent = getS3ObjectContent(payloadPointer);
+
+        LOG.debug("Large message [{}]: {}", getMessageId(message), s3ObjectContent);
```

Comment:
Is it a good idea to dump `s3ObjectContent` to the log - could be huge?

Reply:
it's in debug... if they want to debug. 

Should we truncate and log the first 50 char and latest 50 chars ?